packages/backend/src/nest/registration/registration.service.ts

```diff
@@ -1,178 +1,68 @@
   private readonly logger = Logger(RegistrationService.name)
-  private _server: Server
-  private _port: number
-  public registrationService: any
   public certificates: string[] = []
   private _permsData: PermsData
-  private _ownerCertificate: string
 
-  constructor(@Inject(EXPRESS_PROVIDER) public readonly _app: express.Application) {
+  constructor() {
     super()
   }
 
   onModuleInit() {
-    this.on(RegistrationEvents.SET_CERTIFICATES, certs => {
-      this.setCertificates(certs)
-    })
-    this.on(RegistrationEvents.REGISTER_USER_CERTIFICATE, async (csr: string) => {
-      if (!this._permsData) {
-        console.log('NO PERMS DATA')
-        return
-      }
-      await this.registerUser(csr)
-    })
-    this.setRouting()
-  }
-
-  public setCertificates(certs: string[]) {
-    this.certificates = certs
-  }
-
-  private pendingPromise: Promise<RegistrarResponse> | null = null
-
-  private setRouting() {
-    // @ts-ignore
-    this._app.use(express.json())
-    this._app.post('/register', async (req, res): Promise<void> => {
-      if (this.pendingPromise) return
-      this.pendingPromise = this.registerUser(req.body.data)
-      const result = await this.pendingPromise
-      if (result) {
-        res.status(result.status).send(result.body)
+    this.on(
+      RegistrationEvents.REGISTER_USER_CERTIFICATE,
+      async (payload: { csrs: string[]; certificates: string[] }) => {
+        // Lack of permsData means that we are not the owner of the community
+        if (!this._permsData) return
+        await this.issueCertificates(payload)
       }
-      this.pendingPromise = null
-    })
-  }
-
-  public async listen(): Promise<void> {
-    return await new Promise(resolve => {
-      this._server = this._app.listen(this._port, () => {
-        this.logger(`Certificate registration service listening on port: ${this._port}`)
-        resolve()
-      })
-    })
+    )
   }
 
-  public async stop(): Promise<void> {
-    return await new Promise(resolve => {
-      if (!this._server) resolve()
-      this._server.close(() => {
-        this.logger('Certificate registration service closed')
-        resolve()
-      })
+  private async issueCertificates(payload: { csrs: string[]; certificates: string[] }) {
+    const pendingCsrs = await extractPendingCsrs(payload)
+    pendingCsrs.forEach(async csr => {
+      await this.registerUserCertificate(csr)
     })
   }
 
   public set permsData(perms: PermsData) {
-    console.log('Setting owner perms data')
     this._permsData = {
       certificate: perms.certificate,
       privKey: perms.privKey,
     }
   }
 
   public async registerOwnerCertificate(payload: RegisterOwnerCertificatePayload): Promise<void> {
-    let cert: string
-    try {
-      cert = await registerOwner(payload.userCsr.userCsr, payload.permsData)
-    } catch (e) {
-      this.logger.error(`Registering owner failed: ${e.message}`)
+    // It should not be here.
+    this._permsData = payload.permsData
```

Comment:
Why it's being set here? Did you remove calling LAUNCH_REGISTRAR? I think permsData was set on that event.

Reply:
Yeah and it was too late in case owner registering his own cert - should I figure this out now?

Reply:
Maybe I was not clear: we need both. LAUNCH_REGISTRAR sets permsData if owner already have certificate, if he doesn't have a certificate it's being set when issuing certificate. It should be set by launch_registrar always in my opinion but this involve additional work of figuring out what's wrong in state-manager and make a scope of this PR much bigger possibly, lmk if you want mi to chase it now @leblowl

Reply:
Ok, I think that makes sense. Can we add some more detail to the comment:
```
    // It should not be here.
```
to explain a little more why it shouldn't be there.